src/Microsoft.AspNet.OData.Shared/Formatter/Deserialization/ODataResourceDeserializer.cs

```diff
@@ -206,6 +218,16 @@ public virtual object CreateResourceInstance(IEdmStructuredTypeReference structu
             {
                 if (structuredType.IsEntity())
                 {
+                    if (readContext.IsDeltaDeletedEntity)
+                    {
+                        return new EdmDeltaDeletedEntityObject(structuredType.AsEntity());
+                    }
+
+                    if (readContext.IsDeltaEntity)
+                    {
+                        return new EdmDeltaEntityObject(structuredType.AsEntity());
+                    }
```

Comment:
we should not distiguish DeltaEntity and Normal Entity.
DeltaEntity is a normal entity. @mikepizzo

Reply:
EdmDeltaEntityObject inherits from IEdmChangedObject  (edmchangedobjcoll is coll if idmchangedobject) , dut EdmEntity doesnt

---
In reply to: [604622500](https://github.com/OData/WebApi/pull/2455#discussion_r604622500) [](ancestors = 604622500)